src/big/big_float.cr

```diff
@@ -17,6 +17,47 @@ struct BigFloat < Float
     LibGMP.mpf_init_set_str(out @mpf, str, 10)
   end
 
+  def initialize(num : BigInt)
+    LibGMP.mpf_init(out @mpf)
+    LibGMP.mpf_set_z(self, num)
+  end
+
+  def initialize(num : BigRational)
+    LibGMP.mpf_init(out @mpf)
+    LibGMP.mpf_set_q(self, num)
+  end
+
+  def initialize(num : BigFloat)
+    LibGMP.mpf_init(out @mpf)
+    LibGMP.mpf_set(self, num)
+  end
+
+  def initialize(num : Int8 | Int16 | Int32)
+    LibGMP.mpf_init_set_si(out @mpf, num)
+  end
+
+  def initialize(num : UInt8 | UInt16 | UInt32)
+    LibGMP.mpf_init_set_ui(out @mpf, num)
+  end
+
+  def initialize(num : Int64)
+    if LibGMP::Long == Int64
+      LibGMP.mpf_init_set_si(out @mpf, num)
+    else
+      LibGMP.mpf_init(out @mpf)
+      LibGMP.mpf_set_z(self, num.to_big_i)
+    end
+  end
+
+  def initialize(num : UInt64)
+    if LibGMP::ULong == UInt64
+      LibGMP.mpf_init_set_ui(out @mpf, num)
+    else
+      LibGMP.mpf_init(out @mpf)
+      LibGMP.mpf_set_z(self, num.to_big_i)
+    end
+  end
+
   def initialize(num : Number)
```

Comment:
As I suspected, removing this type restriction fixes the issue.

Reply:
Yes, `def initialize(num)` works.
But it is ridiculous.
Should this method accept anything with `#to_f64` method, or it should accept only Number?
If latter, then this restriction could not be removed.

Reply:
I'd rather merge this with the overloads working and not have to change it later than merge it with the ugly case + comment in initialize. We can reproduce the failure so we can investigate it.

Reply:
Just fix overloads by forward declaration of types as @asterite points to.